srcts/src/shiny/bind.ts

```diff
@@ -85,9 +131,16 @@ function bindInputs(
       if (el.hasAttribute("data-shiny-no-bind-input")) continue;
       const id = binding.getId(el);
 
-      // Check if ID is falsy, or if already bound
-      if (!id || boundInputs[id]) continue;
+      // Check for duplicates in bindingIds array and keep track of them
+      const duplicateId = bindingIdExists(id);
+      if (duplicateId) {
+        inputDuplicateIds.add(id);
+      }
+      // Check if ID is falsy, or if already bound, or has the same ID as
+      // another input binding and if it is, skip
+      if (!id || boundInputs[id] || duplicateId) continue;
```

Comment:
Move `!id` check above the duplicated ID check.

Remove all of the `boundInputs` code because it's redundant with the new code.